packages/app-builder-lib/src/targets/FlatpakTarget.ts

```diff
@@ -0,0 +1,169 @@
+
+  private async copyIcons(stageDir: StageDir) {
+    const icons = await this.helper.icons
+    const copyIcons = icons.map(async icon => {
+      const extWithDot = path.extname(icon.file)
+      const sizeName = extWithDot === ".svg" ? "scalable" : `${icon.size}x${icon.size}`
+      const iconDst = stageDir.getTempFile(path.join("share", "icons", "hicolor", sizeName, "apps", `${this.packager.appInfo.id}${extWithDot}`))
+
+      return copyFile(icon.file, iconDst)
+    })
+
+    await Promise.all(copyIcons)
+  }
+
+  private getFlatpakBuilderOptions(appOutDir: string, stageDir: string, artifactName: string, arch: Arch): { manifest: FlatpakManifest; buildOptions: FlatpakBundlerBuildOptions } {
+    const appIdentifier = this.packager.appInfo.id
+    const { executableName } = this.packager
+    const flatpakArch = toLinuxArchString(arch, "flatpak")
+
+    const manifest: FlatpakManifest = {
+      id: appIdentifier,
+      command: "electron-wrapper",
+      runtime: this.options.runtime || flatpakBuilderDefaults.runtime,
+      runtimeVersion: this.options.runtimeVersion || flatpakBuilderDefaults.runtimeVersion,
+      sdk: this.options.sdk || flatpakBuilderDefaults.sdk,
+      base: this.options.base || flatpakBuilderDefaults.base,
+      baseVersion: this.options.baseVersion || flatpakBuilderDefaults.baseVersion,
+      finishArgs: this.options.finishArgs || flatpakBuilderDefaults.finishArgs,
+      branch: this.options.branch,
+      modules: this.options.modules,
+    }
+
+    const buildOptions: FlatpakBundlerBuildOptions = {
+      baseFlatpakref: `app/${manifest.base}/${flatpakArch}/${manifest.baseVersion}`,
+      runtimeFlatpakref: `runtime/${manifest.runtime}/${flatpakArch}/${manifest.runtimeVersion}`,
+      sdkFlatpakref: `runtime/${manifest.sdk}/${flatpakArch}/${manifest.runtimeVersion}`,
+      arch: flatpakArch as any,
+      bundlePath: path.join(this.outDir, artifactName),
+      files: [[stageDir, "/"], [appOutDir, path.join("/lib", appIdentifier)], ...(this.options.files || [])],
+      symlinks: [[path.join("/lib", appIdentifier, executableName), path.join("/bin", executableName)], ...(this.options.symlinks || [])],
+    }
+
+    return { manifest, buildOptions }
+  }
+
+  private isElectronVersionGreaterOrEqualThan(version: string): boolean {
+    return semver.gte(this.packager.config.electronVersion || "7.0.0", version)
+  }
+}
+
+const flatpakBuilderDefaults: Omit<FlatpakManifest, "id" | "command"> = {
+  runtime: "org.freedesktop.Platform",
+  runtimeVersion: "20.08",
+  sdk: "org.freedesktop.Sdk",
+  base: "org.electronjs.Electron2.BaseApp",
+  baseVersion: "20.08",
+  finishArgs: [
+    // Wayland
+    "--socket=wayland",
+    // X11
+    "--socket=x11",
+    "--share=ipc",
+    // Open GL
+    "--device=dri",
+    // Audio output
+    "--socket=pulseaudio",
+    // Read/write home directory access
+    "--filesystem=home",
+    // Chromium uses a socket in tmp for its singleton check
+    "--env=TMPDIR=/var/tmp",
```

Comment:
Note that most Flatpaks instead set `TMPDIR=$XDG_RUNTIME_DIR/app/$FLATPAK_ID` instead.

Reply:
Nice catch! Initially I've copied these values from [@malept/electron-installer-flatpak](https://github.com/malept/electron-installer-flatpak/blob/097e08654dfc8d0eeeee1ad3ed617e17e607c0b8/README.md#optionsfinishargs) without giving it too much thought.

But now that you've mentioned I've spent some time looking into it and I can't seem to find these chromium socket files anywhere (neither in `/tmp` nor in `$XDG_RUNTIME_DIR`). So now I'm wondering whether this config is still needed or is it just some artifact of the past where the documentation hasn't kept up with the latest changes.

So unless someone can confirm that this config is still needed I'm tempted to just remove it (0400b402da6174e4a872f4e3a11caa53baee2db1) and wait until there's a compelling reason to bring it back.

Reply:
Try running with `ls -a`, the files are often hidden by default. At minimum, I want to say it's at least a pretty decent chunk of Electron apps that do this.

Reply:
You're right! It turns out the `TMPDIR` is actually used by Electron for creating the singleton socket.

However, this socket is only created when the application is using the [app.apprequestsingleinstancelock](https://www.electronjs.org/docs/api/app#apprequestsingleinstancelock) API. My previous tests were done using a sample application which wasn't using this API and that's why the socket was missing.

Therefore I've pushed another fix (https://github.com/electron-userland/electron-builder/pull/5711/commits/9ffc66f8da44c9ffc55006ccc0ae62e12e5a394c) that sets `TMPDIR=$XDG_RUNTIME_DIR/app/$FLATPAK_ID` in the wrapper script. I've done a small test and the socket file is indeed created in the right place (`$XDG_RUNTIME_DIR/app/$FLATPAK_ID`) and it gets cleaned up properly when the last instance of the app gets closed.